panoramix/decompiler.py

```diff
@@ -155,7 +155,7 @@ def _decompile_with_loader(loader, only_func_name=None) -> Decompilation:
             if target > 1 and loader.lines[target][1] == "jumpdest":
                 target += 1
 
-            @timeout_decorator.timeout(60 * 3, timeout_exception=TimeoutInterrupt)
+            @timeout(60 * 3, timeout_exception=TimeoutInterrupt, use_signals=False)
```

Comment:
Performance wise, should this be made conditionnal based on the plateform not being Unix ?

Reply:
I need it in Linux Ubuntu and in Macos because I use panoramix in a separate thread

Reply:
@NikZak : then the cost of using detecting if it’s not in the main thread is very low.

Reply:
I agree. Although I reckon initializing a process is a miniscule part of the whole decompiling procedure.

Reply:
I think that writing something like "if platform is windows or if runing not in the main thread" is brittle and kind of ugly.

Reply:
No. It’s something wanted for large contracts.